src/main/java/net/snowflake/client/core/SFArrowResultSet.java

```diff
@@ -559,14 +560,17 @@ public Timestamp getTimestamp(int columnIndex, TimeZone tz) throws SFException {
 
   @Override
   public Object getObject(int columnIndex) throws SFException {
+    int type = resultSetMetaData.getColumnType(columnIndex);
+    if (type == SnowflakeUtil.EXTRA_TYPES_VECTOR) {
+      return getString(columnIndex);
```

Comment:
Why so? Why not return `[]int` or `[]float`?

Reply:
Vector is like a Structured Types - we have specific methods to extract the vector as `int[]` or `float[]` e.g. 
```
Integer[] intVector = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1,` Integer.class);
Float[] result = resultSet.unwrap(SnowflakeBaseResultSet.class).getArray(1, Float.class);
```
but when `getObject` is called than we return its string representation.